### PR TITLE
    doc: Updated documentation for CS2 release

### PR DESCRIPTION
**Pull request by FrancescoSer**
doc: Updated documentation for CS2 release

    Updated 54H custom pcb guide.
    Updated links.txt
    Updated 54H gs guide.
    Added CS2 migration notes.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/migration/migration_guide_nRF54H20_cs_to_2_7_99-cs2.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+nRF54H20 BICR
+  The nRF54H20 BICR has been updated.
+
+  To update the BICR of your development kit while in Root of Trust, do the following:
+
+  1. Download the `BICR new binary file`_.
+  #. Connect the nRF54H20 DK to your computer using the **DEBUGGER** port on the DK.
+
+     .. note::
+        On MacOS, connecting the DK might cause a popup containing the message ``Disk Not Ejected Properly`` to repeatedly appear on screen.
```
Comment (annwoj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_nRF54H20_cs_to_2_7_99-cs2.rst:47]:
```suggestion
        On MacOS, connecting the DK may repeatedly trigger a popup displaying the message ``Disk Not Ejected Properly``.
```

File: doc/nrf/releases_and_maturity/abi_compatibility.rst [hunk 2]
```diff
@@ -28,6 +28,8 @@ The following table illustrates ABI compatibility between different versions of
 
    * - |NCS| versions
      - Compatible nrf54h20_soc_binary version
+   * - |NCS| v2.7.99-cs2
+     - nrf54h20_soc_binaries v0.6.4
```
Comment (FrancescoSer) [inline doc/nrf/releases_and_maturity/abi_compatibility.rst:32]: Verify final name, add linking (?)

File: doc/nrf/releases_and_maturity/migration/migration_guide_nRF54H20_cs_to_2_7_99-cs2.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+        * From the Command Prompt::
+
+          JLink_Windows_V794i_x86_64.exe -InstUSBDriver=1
+
+        * From PowerShell::
+
+          .\JLink_Windows_V794i_x86_64.exe -InstUSBDriver=1
+
+    #. In the :guilabel:`Choose optional components` window, select again :guilabel:`update existing installation`.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/migration/migration_guide_nRF54H20_cs_to_2_7_99-cs2.rst:91]: Why is this "again"? I don't see the option selected in any other step.

**Review by wiba-nordic — APPROVED** [review]

**Review by annwoj — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Comment by gmarull** [comment]
```
/home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/releases_and_maturity/migration/migration_guide_nRF54H20_cs_to_2_7_99-cs2.rst:76: ERROR: Unknown target name: "j-link version 7.94i".
```

**Comment by wiba-nordic** [comment]
> ```
> /home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/releases_and_maturity/migration/migration_guide_nRF54H20_cs_to_2_7_99-cs2.rst:76: ERROR: Unknown target name: "j-link version 7.94i".
> ```

J-Link linking style changed by https://github.com/nrfconnect/sdk-nrf/pull/17348/files

**Pull request merged**